packages/x-components/src/x-modules/next-queries/store/actions/fetch-and-save-next-query-preview.action.ts

```diff
@@ -0,0 +1,43 @@
+import { NextQueriesXStoreModule } from '../types';
+import { cancellablePromise, CancelSymbol } from '../../../../utils/cancellable-promise';
+
+let cancelPromise: undefined | (() => void);
```

Comment:
Like below, multiple requests can happen at the same time, and they will override this variable, so this global variable for me does not make sense. 
Probably we can skip cancellation handling for now and simplify this code. I wasn't thinking about adding it when I refined the task, as the worst think that can happen is that you have an additional entry of previews stored. Maybe there are some corner cases like extra parameters that change while you are requesting the preview, but I don't think it is enough to complicate the code right now.